src/NuGet.Core/NuGet.Frameworks/NuGetFrameworkFactory.cs

```diff
@@ -92,18 +92,19 @@ public static NuGetFramework ParseFrameworkName(string frameworkName, IFramework
             // if the first part is a special framework, ignore the rest
             if (!TryParseSpecialFramework(parts[0], out result))
             {
```

Comment:
Just in case add `parts.Any()` if expecting at least 1 item in array which is parts[0] for not to get index out of range exception.
`if (parts.Any() && !TryParseSpecialFramework(parts[0], out result))`

Reply:
When you know the data type, using `.Length > 0` or `.Count > 0` is a lot faster than using LINQ to do the same. Of course, this one thing is such a small part of the overall execution that in customer experience % terms it'll be just about zero, but I personally encourage people to avoid LINQ where possible.